src/game_battlealgorithm.cpp

```diff
@@ -1322,22 +1323,33 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 		for (int i = 0; i < (int) skill.state_effects.size(); i++) {
 			if (!skill.state_effects[i])
 				continue;
+			auto state_id = i + 1;
 
-			if (!healing && !Utils::PercentChance(to_hit))
+			if (!healing && GetTarget()->HasState(state_id)) {
+				this->success = true;
+				conditions.push_back(state_id);
+				continue;
+			}
+			if (healing && !GetTarget()->HasState(state_id)) {
+				continue;
+			}
+			if (!Utils::PercentChance(to_hit))
```

Comment:
did you verify that a hit test really applies to headling skills?

Reply:
okay, just tested, is correct oO

Reply:
Tested it also